crates/gen-host-js/src/lib.rs

```diff
@@ -368,8 +398,26 @@ impl Js {
                 const dataView = mem => dv.buffer === mem.buffer ? dv : dv = new DataView(mem.buffer);
             "),
 
+            Intrinsic::LoadWasm => if self.opts.nodejs_compat {
+                self.src.js("
+                    const isNode = typeof process !== 'undefined' && process.versions && process.versions.node;
+                    let _fs;
+                    async function loadWasm (url) {
+                        if (isNode) {
+                            _fs = _fs || await import('node:fs/promises');
+                            return WebAssembly.compile(await _fs.readFile(url));
+                        }
+                        return fetch(url).then(WebAssembly.compile);
+                    }
```

Comment:
Well spotted. In theory you avoid one event loop tick by awaiting the promise in the async function instead of returning it. But it's completely negligible from any user perspective so that writing the shortest code is still the preference for me with these. Changing to be a non-async function for now.

Reply:
Actually the `import('node:fs')` does need the await, so I've left it as-is for now.